shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm

```diff
@@ -220,57 +197,513 @@ - (CGRect)accessibilityFrame {
+  if ([self node].HasAction(blink::SemanticsAction::kIncrease)) {
+    [self node].value = [self node].increasedValue;
+    [self bridge]->DispatchSemanticsAction([self uid], blink::SemanticsAction::kIncrease);
   }
 }
 
 - (void)accessibilityDecrement {
-  if (_node.HasAction(blink::SemanticsAction::kDecrease)) {
-    _node.value = _node.decreasedValue;
-    _bridge->DispatchSemanticsAction(_uid, blink::SemanticsAction::kDecrease);
+  if ([self node].HasAction(blink::SemanticsAction::kDecrease)) {
+    [self node].value = [self node].decreasedValue;
+    [self bridge]->DispatchSemanticsAction([self uid], blink::SemanticsAction::kDecrease);
   }
 }
 
 - (BOOL)accessibilityScroll:(UIAccessibilityScrollDirection)direction {
   blink::SemanticsAction action = GetSemanticsActionForScrollDirection(direction);
-  if (!_node.HasAction(action))
+  if (![self node].HasAction(action))
     return NO;
-  _bridge->DispatchSemanticsAction(_uid, action);
+  [self bridge]->DispatchSemanticsAction([self uid], action);
   return YES;
 }
 
 @end
 
+@interface InactiveTextPosition : UITextPosition
+
+@property(nonatomic, readonly) NSUInteger index;
+
++ (instancetype)positionWithIndex:(NSUInteger)index;
+- (instancetype)initWithIndex:(NSUInteger)index;
+
+@end
+
+@implementation InactiveTextPosition
+
++ (instancetype)positionWithIndex:(NSUInteger)index {
+  return [[[InactiveTextPosition alloc] initWithIndex:index] autorelease];
+}
+
+- (instancetype)initWithIndex:(NSUInteger)index {
+  self = [super init];
+  if (self) {
+    _index = index;
+  }
+  return self;
+}
+
+@end
+
+@interface InactiveTextRange : UITextRange<NSCopying>
+
+@property(nonatomic, readonly) NSRange range;
+
++ (instancetype)rangeWithNSRange:(NSRange)range;
+
+@end
+
+@implementation InactiveTextRange
+
++ (instancetype)rangeWithNSRange:(NSRange)range {
+  return [[[InactiveTextRange alloc] initWithNSRange:range] autorelease];
+}
+
+- (instancetype)initWithNSRange:(NSRange)range {
+  self = [super init];
+  if (self) {
+    _range = range;
+  }
+  return self;
+}
+
+- (UITextPosition*)start {
+  return [InactiveTextPosition positionWithIndex:self.range.location];
+}
+
+- (UITextPosition*)end {
+  return [InactiveTextPosition positionWithIndex:self.range.location + self.range.length];
+}
+
+- (BOOL)isEmpty {
+  return self.range.length == 0;
+}
+
+- (id)copyWithZone:(NSZone*)zone {
+  return [[InactiveTextRange allocWithZone:zone] initWithNSRange:self.range];
+}
+
+@end
+
+@implementation InactiveTextInput {
+}
+
+@synthesize tokenizer = _tokenizer;
+@synthesize beginningOfDocument = _beginningOfDocument;
+@synthesize endOfDocument = _endOfDocument;
+
+- (instancetype)init {
+  return [super init];
+}
+
+- (BOOL)hasText {
+  return self.text.length > 0;
+}
+
+- (NSString*)textInRange:(UITextRange*)range {
+  NSRange textRange = ((InactiveTextRange*)range).range;
+  return [self.text substringWithRange:textRange];
+}
+
+- (void)replaceRange:(UITextRange*)range withText:(NSString*)text {
```

Comment:
Can we add comments to the empty and placeholder implementations to indicate if they have been left blank intentionally or if filling them in is a TODO?

Reply:
Done.